### PR TITLE
Update rollout operator to 0.8.1

### PR DESCRIPTION
**Pull request by jhalterman**
#### What this PR does

Updates Mimir to use rollout-operator 0.8.1, via the 0.9.1 helm chart.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by andyasp — APPROVED** [review]
LGTM

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 
 ## main / unreleased
 
+* [ENHANCEMENT] Update the `rollout-operator` subchart to `0.9.1`. #6110
 * [ENHANCEMENT] Update the `rollout-operator` subchart to `0.9.0`. #6022
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:32]: should we collapse the two entries into one?

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 * [ENHANCEMENT] Double the amount of rule groups for each user tier. #5897
 * [ENHANCEMENT] Set `maxUnavailable` to 0 for `distributor`, `overrides-exporter`, `querier`, `query-frontend`, `query-scheduler` `ruler-querier`, `ruler-query-frontend`, `ruler-query-scheduler` and `consul` deployments, to ensure they don't become completely unavailable during a rollout. #5924
-* [ENHANCEMENT] Update rollout-operator to `v0.8.0`. #6022
+* [ENHANCEMENT] Update rollout-operator to `v0.8.1`. #6022 #6110
```
Comment (dimitarvdimitrov) [inline CHANGELOG.md:29]: you probably meant 0.9.1, not 0.8.1?

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, with two nitpicks on the changelog entries

File: operations/helm/charts/mimir-distributed/CHANGELOG.md (hunk 1 above)
Comment (andyasp) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md:32]:
```suggestion
* [ENHANCEMENT] Update the `rollout-operator` subchart to `0.9.1`. #6022 #6110
```

**Pull request merged**